packages/haul-core/src/webpack/plugins/RamBundlePlugin/RamBundlePlugin.ts

```diff
@@ -0,0 +1,308 @@
+        mainChunk = chunk;
+        mainId = chunk.entryModule?.id ?? undefined;
+      }
+
+      chunk.getModules().forEach(moduleInChunk => {
+        if (chunk.id !== null) {
+          moduleMappings.chunks[chunk.id] = ([] as Array<string | number>)
+            .concat(...(moduleMappings.chunks[chunk.id] || []))
+            .concat(moduleInChunk.id ?? []);
+        }
+      });
+    });
+
+    // Omit chunks mapping if there's only a single main chunk
+    if (compilation.chunks.length === 1) {
+      moduleMappings.chunks = {};
+    }
+
+    if (mainId === undefined) {
+      throw new Error(
+        "RamBundlePlugin: couldn't find main chunk's entry module id"
+      );
+    }
+    if (!mainChunk) {
+      throw new Error("RamBundlePlugin: couldn't find main chunk");
+    }
+    // Render modules to it's 'final' form with injected webpack variables
+    // and wrapped with ModuleTemplate.
+    const minifyWorker = new Worker(
+      require.resolve(
+        '../../../../build/webpack/plugins/RamBundlePlugin/minifyWorker'
+      ),
+      {
+        numWorkers: this.config.maxWorkers,
+        enableWorkerThreads: true,
+      }
+    );
+
+    this.modules = await Promise.all(
+      compilation.modules.map(async webpackModule => {
+        const renderedModule = compilation.moduleTemplates.javascript
+          .render(
+            webpackModule,
+            compilation.dependencyTemplates,
+            compilation.options
+          )
+          .sourceAndMap();
+
+        if (typeof webpackModule.id === 'string') {
+          moduleMappings.modules[webpackModule.id] = webpackModule.index;
+        }
+
+        let code = `__haul_${this.config.bundleName}.l(${variableToString(
+          webpackModule.id
+        )}, ${renderedModule.source});`;
+        let map = renderedModule.map;
+
+        if (this.config.minify) {
+          const minifyOptionsWithMap = {
+            ...this.config.minifyOptions,
+            sourceMap: {
+              content: map,
+            },
+          };
+          // @ts-ignore property minify does not exist on type 'JestWorker'
+          const minifiedSource = await minifyWorker.minify(
+            code,
+            minifyOptionsWithMap
+          );
+          //Check if there is no error in minifed source
+          assert(!minifiedSource.error, minifiedSource.error);
+
+          code = minifiedSource.code || '';
+          if (typeof minifiedSource.map === 'string') {
+            map = JSON.parse(minifiedSource.map);
+          }
+        }
+
+        return {
+          id: webpackModule.id,
+          idx: webpackModule.index,
+          filename: webpackModule.resource,
+          source: code,
+          map: {
+            ...map,
+            file: `${
+              typeof webpackModule.id === 'string'
+                ? webpackModule.index
+                : webpackModule.id
+            }.js`,
+          },
+        };
+      })
+    );
+
+    minifyWorker.end();
```

Comment:
`end()` is now async: https://github.com/facebook/jest/blob/master/packages/jest-worker/src/index.ts#L140, make sure to check if the version you're on is async as well and if not, mark as todo or something

Reply:
I had a problem previosuly, when awaiting `end()` would never resolve and the whole process would hang indefinitely.